### PR TITLE
fix(ui): RadioButton Tooltip and UI discrepancy [LW-9828, LW-10070]

### PR DESCRIPTION
**Pull request by tommayeliog**
# Checklist

- [x] JIRA - [LW-9828](https://input-output.atlassian.net/browse/LW-9828) and [LW-10070](https://input-output.atlassian.net/browse/LW-10070)
- [x] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

RadioButton component and Story has been refactored so `label` accepts string or ReactNode in order to prevent adding ToolTip to RadioButton. Now it can be any ReactNode. The `icon` is still a property of RadioButton component because it's heavily styled in different states and mouse events.

## Testing
Storybook

## Screenshots
Screenshot 2024-03-18 at 15 57 53
Screenshot 2024-03-18 at 15 58 00
Screenshot 2024-03-18 at 15 58 42
Screenshot 2024-03-18 at 15 58 33
Screenshot 2024-03-18 at 15 58 51
Screenshot 2024-03-18 at 15 58 58

[LW-9828]: https://input-output.atlassian.net/browse/LW-9828?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LW-10070]: https://input-output.atlassian.net/browse/LW-10070?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/ui/src/design-system/radio-button/radio-button.component.tsx [hunk 1]
```diff
@@ -5,13 +5,12 @@ import cn from 'classnames';
...
 import { Box } from '../box';
 import { Flex } from '../flex';
-import { Tooltip } from '../tooltip';
 
 import * as cx from './radio-button.css';
 
 export interface RadioButtonGroupOption {
   value: string;
-  label: string;
+  label: React.ReactNode | string;
```
Comment (przemyslaw-wlodek) [inline packages/ui/src/design-system/radio-button/radio-button.component.tsx:13]: `string` is covered by `ReactNode`

File: packages/ui/src/design-system/radio-button/radio-button.component.tsx [hunk 2]
```diff
@@ -96,14 +86,14 @@ export const RadioButtonGroup = ({
                       disabled={disabled}
                       onClick={onIconClick}
                     >
-                      <Icon />
+                      <Icon width={24} height={24} />
```
Comment (przemyslaw-wlodek) [inline packages/ui/src/design-system/radio-button/radio-button.component.tsx:89]: Please use `className` e.g. `iconButton`

File: packages/ui/src/design-system/radio-button/radio-button.css.ts [hunk 3]
```diff
@@ -100,23 +93,34 @@ export const radioGroupIndicator = style([
...
-      alignSelf: 'center',
-      width: vars.spacing.$8,
-      height: vars.spacing.$8,
-      borderRadius: '50%',
-      backgroundColor: vars.colors.$radiobutton_indicator_check_color,
+    selectors: {
+      '&:active::after': {
+        content: '',
+        display: 'flex',
+        alignSelf: 'center',
```
Comment (przemyslaw-wlodek) [inline packages/ui/src/design-system/radio-button/radio-button.css.ts:100]: Could you please check if it's not a leftover?

File: packages/ui/src/design-system/radio-button/radio-button.css.ts [hunk 4]
```diff
@@ -100,23 +93,34 @@ export const radioGroupIndicator = style([
...
+        width: vars.spacing.$8,
+        height: vars.spacing.$8,
+        borderRadius: '50%',
+        backgroundColor: vars.colors.$radiobutton_indicator_check_color,
+      },
+      '&::after': {
+        content: '',
+        display: 'flex',
+        alignSelf: 'center',
```
Comment (przemyslaw-wlodek) [inline packages/ui/src/design-system/radio-button/radio-button.css.ts:109]: ditto

File: packages/ui/src/design-system/radio-button/radio-button.css.ts [hunk 5]
```diff
@@ -125,29 +129,33 @@ export const iconWrapper = style([
...
 export const iconButton = style([
   sx({
-    backgroundColor: '$transparent',
     borderRadius: '$small',
-    ml: '$24',
-    color: '$radiobutton_icon_color',
+    color: '$radiobutton_icon_text_color',
+    width: '$40',
+    height: '$40',
+    paddingTop: '$4',
```
Comment (przemyslaw-wlodek) [inline packages/ui/src/design-system/radio-button/radio-button.css.ts:136]: Why this strange padding is needed?

File: packages/ui/src/design-system/radio-button/radio-button.css.ts [hunk 6]
```diff
@@ -125,29 +129,33 @@ export const iconWrapper = style([
...
+    backgroundColor: '$radiobutton_icon_color',
   }),
   {
-    padding: '6px 6px 0 6px',
-    border: `1px solid ${vars.colors.$radiobutton_unchecked_borderColor}`,
-
+    border: `1.5px solid ${vars.colors.$radiobutton_icon_hover_border_color}`,
```
Comment (przemyslaw-wlodek) [inline packages/ui/src/design-system/radio-button/radio-button.css.ts:140]: We should use `1px` or `2px`. Please consult with the designer. We agreed some time ago we're not using half-pixels.

(there are more occurrences in this file)

File: packages/ui/src/design-system/radio-button/radio-button.stories.tsx [hunk 7]
```diff
@@ -88,7 +85,9 @@ const MainComponents = (): JSX.Element => {
             {index < 4 ? (
               <RadioButtonGroup {...item} />
             ) : (
-              <Text.Body.Small>* only checked state supported</Text.Body.Small>
+              <Text.Body.Normal className={styles.note}>
```
Comment (przemyslaw-wlodek) [inline packages/ui/src/design-system/radio-button/radio-button.stories.tsx:88]: Please add a TODO to replace it with `<Text.* color="secondary">` after https://input-output.atlassian.net/browse/LW-9760. Ask Dan to update the color if necessary.

**Review by przemyslaw-wlodek — APPROVED** [review]
LGTM Tom :) Left a couple of minor comments.

File: packages/ui/src/design-system/radio-button/radio-button.component.tsx (hunk 1 above)
Comment (tommayeliog) [reply packages/ui/src/design-system/radio-button/radio-button.component.tsx:13]: removed

File: packages/ui/src/design-system/radio-button/radio-button.component.tsx (hunk 2 above)
Comment (tommayeliog) [reply packages/ui/src/design-system/radio-button/radio-button.component.tsx:89]: done in iconButton with fontSize: 24

File: packages/ui/src/design-system/radio-button/radio-button.css.ts (hunk 3 above)
Comment (tommayeliog) [reply packages/ui/src/design-system/radio-button/radio-button.css.ts:100]: removed

File: packages/ui/src/design-system/radio-button/radio-button.css.ts (hunk 4 above)
Comment (tommayeliog) [reply packages/ui/src/design-system/radio-button/radio-button.css.ts:109]: removed

File: packages/ui/src/design-system/radio-button/radio-button.css.ts (hunk 5 above)
Comment (tommayeliog) [reply packages/ui/src/design-system/radio-button/radio-button.css.ts:136]: removed, I've added align-items: center, justify-content: center to parent

File: packages/ui/src/design-system/radio-button/radio-button.css.ts (hunk 6 above)
Comment (tommayeliog) [reply packages/ui/src/design-system/radio-button/radio-button.css.ts:140]: hmmm, I changed this to 2px for the icon and 1px for the indicator.

File: packages/ui/src/design-system/radio-button/radio-button.stories.tsx (hunk 7 above)
Comment (tommayeliog) [reply packages/ui/src/design-system/radio-button/radio-button.stories.tsx:88]: I've added TODO, and ask Dan to change to color in Figma

**Review by wklos-iohk — APPROVED** [review]

**Review by mchappell — APPROVED** [review]
:shipit: Nice work!

**Pull request merged**